atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/Transaction.java

```diff
@@ -41,25 +41,95 @@
+     * should ONLY be performed through the original byte arrays that were passed in.
+     *
+     * @param tableRef table to load values from
+     * @param rows unique rows to apply the column range selection to
+     * @param columnRangeSelection range of columns and batch size to load for each of the rows provided
+     * @return a mapping of rows to cells matching the predicate in the row, following the ordering outlined above
+     * @throws IllegalArgumentException if {@code rows} contains duplicates
+     */
     @Idempotent
     Map<byte[], BatchingVisitable<Map.Entry<Cell, byte[]>>> getRowsColumnRange(
             TableReference tableRef,
             Iterable<byte[]> rows,
             BatchColumnRangeSelection columnRangeSelection);
 
+    /**
+     * Returns a single iterator over the cell-value pairs in {@code tableRef} for the specified {@code rows}, where the
+     * columns fall within the provided {@link ColumnRangeSelection}. The single provided {@link ColumnRangeSelection}
+     * applies to all of the rows.
+     *
+     * If the provided {@link Iterable} of {@code rows} has a stable ordering, the returned iterator is guaranteed
+     * to return cell-value pairs in a lexicographic ordering over rows and columns, where rows are sorted according to
+     * the stable ordering of {@code rows}, and columns are sorted on byte ordering. If {@code rows} does not have a
+     * stable ordering, behaviour is undefined.
+     *
+     * @param tableRef table to load values from
+     * @param rows unique rows to apply the column range selection to
+     * @param columnRangeSelection range of columns to load for each of the rows provided
+     * @param batchHint number of columns that should be loaded from the underlying database at once
+     * @return an iterator over cell-value pairs, guaranteed to follow the ordering outlined above
+     * @throws IllegalArgumentException if {@code rows} contains duplicates
+     */
     @Idempotent
     Iterator<Map.Entry<Cell, byte[]>> getRowsColumnRange(
             TableReference tableRef,
             Iterable<byte[]> rows,
             ColumnRangeSelection columnRangeSelection,
             int batchHint);
 
+    /**
+     * Returns a mapping of rows to {@link Iterator}s over cell-value pairs within {@code tableRef} for the specified
+     * {@code rows}, where the columns fall within the provided {@link BatchColumnRangeSelection}. The single provided
+     * {@link BatchColumnRangeSelection} applies to all of the rows.
+     *
+     * The returned {@link Iterator}s are guaranteed to return cells matching the predicate, sorted on (ascending) byte
+     * ordering.
+     *
+     * It is guaranteed that the {@link Map#keySet()} of the returned map has the same elements as {@code rows} by
+     * identity, even if there are rows where no columns match the predicate. Random access for the returned map
+     * should ONLY be performed through the original byte arrays that were passed in.
```

Comment:
Likewise. Basically, if we get each method to the point where you can always lookup in the map via any equivalent byte array, then we can just shove it in a single comment at the top and avoid it on each method.

Reply:
As above: this was done.